packages/beacon-node/src/chain/opPools/opPool.ts

```diff
@@ -316,8 +323,17 @@ export class OpPool {
    * Prune if validator has already exited at or before the finalized checkpoint of the head.
    */
   private pruneVoluntaryExits(headState: CachedBeaconStateAllForks): void {
+    const {config} = headState;
+    const headStateFork = config.getForkSeq(headState.slot);
```

Comment:
Shouldn't we be deleting exits later than the fork based on clock slot? I guess it doesn't much matter if we delete them eventually anyway and they're filtered out during block building.

Reply:
Yes, but using the head state makes the API simpler, and functionally quasi equivalent. The only difference is that exists may be pruned a bit latter in case of fully skipped epochs.